test/gtest/ucs/test_memtype_cache.cc

```diff
@@ -20,22 +26,33 @@ class test_memtype_cache : public ucs::test_with_param<ucs_memory_type_t> {
         ucs::test_with_param<ucs_memory_type_t>::init();
         ucs_status_t status = ucs_memtype_cache_create(&m_memtype_cache);
         ASSERT_UCS_OK(status);
+
+        m_type = GetParam();
     }
 
     virtual void cleanup() {
         ucs_memtype_cache_destroy(m_memtype_cache);
         ucs::test_with_param<ucs_memory_type_t>::cleanup();
     }
 
-    void test_lookup_found(void *ptr, size_t size) {
+    void test_lookup_found(void *ptr, size_t size,
+                           ucs_memory_type_t expected_type = m_type) const {
```

Comment:
why do we need to save m_type? seems test_lookup_found is always called with memory type parameter (from test_ptr_found)

Reply:
good catch - done